hbase-server/src/main/java/org/apache/hadoop/hbase/coordination/ZKSplitTransactionCoordination.java

```diff
@@ -40,6 +40,9 @@
   private CoordinatedStateManager coordinationManager;
   private final ZooKeeperWatcher watcher;
 
+  // max wait for split transaction - 100 times in a loop with 100 ms of thread sleep each time
+  private static final int SPLIT_WAIT_TIMEOUT = 100;
```

Comment:
10s seems rather small for a busy cluster, no? Concern is that it might result in unnecessary abort splits and rollbacks.

Reply:
this accounts for ~24 s due to calls involved in loop. even for busy cluster, by this time, we should be able to retry calling setData() to ZK at least 10 times which I think should be sufficient if ZK node version is changed. In fact, ideally, 2nd time only should be sufficient to retrieve correct ZK node version and successfully updating RIT info in ZK node.

Reply:
I see, ok.

Reply:
I'm going to put this response into the code here as a comment because I was about to request changes on this point.